From fd7547249b57d5c34ae0ed4d0269ad1e34df56f1 Mon Sep 17 00:00:00 2001
From: Kareem Khazem <karkhaz@amazon.com>
Date: Thu, 31 Jan 2019 15:07:52 -0800
Subject: [PATCH 2/3] Change FreeRTOS_IP_Private.h union to struct

This patch should be removed soon: It is for a CBMC issue being fixed now.
---
 lib/FreeRTOS-Plus-TCP/include/FreeRTOS_IP_Private.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/FreeRTOS-Plus-TCP/include/FreeRTOS_IP_Private.h b/lib/FreeRTOS-Plus-TCP/include/FreeRTOS_IP_Private.h
index 6006f89f0..d1a0cf898 100644
--- a/lib/FreeRTOS-Plus-TCP/include/FreeRTOS_IP_Private.h
+++ b/lib/FreeRTOS-Plus-TCP/include/FreeRTOS_IP_Private.h
@@ -643,7 +643,7 @@ typedef struct XSOCKET
 	/* Before accessing any member of this structure, it should be confirmed */
 	/* that the protocol corresponds with the type of structure */
 
-	union
+	struct
 	{
 		IPUDPSocket_t xUDP;
 		#if( ipconfigUSE_TCP == 1 )
-- 
2.21.0

